tx_test.go

```diff
@@ -139,6 +142,68 @@ func TestAddInputFromTx(t *testing.T) {
 	assert.Equal(t, newTx.GetTotalInputSatoshis(), uint64(200000))
 }
 
+// This test reads a sample block from a file, but normally
+// we would get the block directly from the node via a REST GET...
+/*
+	resp, err := http.Get(fmt.Sprintf("%s/rest/block/%s.bin", b.client.serverAddr, blockHash))
+	if err != nil {
+		return nil, fmt.Errorf("Could not GET block: %v", err)
+	}
+	if resp.StatusCode != 200 {
+		defer resp.Body.Close()
+		data, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			return nil, fmt.Errorf("failed to read response body: %w", err)
+		}
+		return nil, fmt.Errorf("ERROR: code %d: %s", resp.StatusCode, data)
+	}
+	Then use resp.Body as in the test below
+*/
+func TestNewTxFromReader(t *testing.T) {
+	var err error
+
+	f, err := os.Open("./block.bin")
+	if err != nil {
+		t.Error(err)
+		t.FailNow()
+	}
+	defer f.Close()
+
+	r := bufio.NewReader(f)
+
+	header := make([]byte, 80)
+	if n, err := io.ReadFull(f, header); n != 80 || err != nil {
```

Comment:
@ordishs can you change this to:
```
	var n int
	if n, err = io.ReadFull(f, header); n != 80 || err != nil {
```
because when running `make test-ci` it's causing the linter to fail here:
```
tx_test.go:175:8: shadow: declaration of "err" shadows declaration at line 163 (govet)
	if n, err := io.ReadFull(f, header); n != 80 || err != nil {
	      ^
```

Reply:
Fixed